#175636847 Add routes for getting and updating profile information

hervebutera opened the pull request:
#### What does this PR do?
Adds functionality for getting user information by id.
Adds functionality for updating user information.
#### Description of Task to be completed?
- When the user is logged in, he/she can get his/her profile
information by id.
- The logged-in user also is able to update his profile information.
- The user is also able to change his profile picture which is uploaded on Cloudinary.
#### How should this be manually tested?
- clone the repo by `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- run `cd tech-strikers-bn-backend`
- run `git checkout ft-profile-page-settings-175636847`
- run `npm install`
- add `CLOUDINARY_NAME=` in .env file and corresponding value of your cloudinary account for testing.
- add `CLOUDINARY_API_KEY=` in .env file and corresponding value of your cloudinary account for testing.
- add `CLOUDINARY_API_SECRET=` in .env file and corresponding value of your cloudinary account for testing.
- In your cloudinary account settings, go to upload settings, create an `upload preset` with mode `Unsigned`.
- Also in `src/controllers/user.controller.js`, add the created upload preset name in the second parameter 
of  `uploadProfilePic` function.
- run `npm run dev`
- use postman to test `GET  http://localhost:3000/api/user/<any user id>`
-use postman to test  `PUT http://localhost:3000/api/user/<any user id>` 
#### Any background context you want to provide?
Cloudinary offers limited storage GB for a free account so every developer has to use his cloudinary account for
development and testing purposes.
Also a default value of a cloudinary image has been added in the `profilePicture` column for every user.
#### What are the relevant pivotal tracker stories?
[#175636847](https://www.pivotaltracker.com/story/show/175636847)
#### Screenshots (if appropriate)
- Postman screenshot of `GET` route for user table.
<img width="804" alt="Capture1" src="https://user-images.githubusercontent.com/48241489/102896034-6b2af200-446e-11eb-94ab-4169c389ee24.PNG">

- Postman screenshot of unsuccessful `PUT` operation when `username` has been taken
<img width="801" alt="Capture2" src="https://user-images.githubusercontent.com/48241489/102895846-21420c00-446e-11eb-8e38-2a51084fb0ce.PNG">

- Postman screenshot of successful `PUT` route (updating user information)for user 
<img width="811" alt="Capture" src="https://user-images.githubusercontent.com/48241489/102895727-e50eab80-446d-11eb-921d-de40d8ebad2d.PNG">



#### Questions:

tytyne reviewed (changes requested):
I thought changing profile, we had to add picture  and other information related to the user!

Duncanwar commented on src/controllers/user.controller.js:
```diff
@@ -0,0 +1,26 @@
+
+
+
+export default { getUserInfo, upDateUser };
```
Can you transform your functions into a class?
For readability and solid principles

Duncanwar commented on src/services/user.service.js:
```diff
@@ -0,0 +1,27 @@
+    }}
+} 
+
+export default { retrieveUser, upDateUserInfo }; 
```
We can use a class

Duncanwar commented on src/routes/api/users.js:
```diff
@@ -1,29 +1,10 @@
-});
+router.put("/user/:userId", upDateUser);
 
 export default router;
```
You don't have to remove the above lines.

Duncanwar reviewed (commented):
Implement those feedback

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/atlp-rwanda/tech-strikers-bn-backend/837527c582d32c231319de91fab09efa44b1accc/atlp-rwanda/tech-strikers-bn-backend/7e55e2edc3dba77480ef22a969249d63f5609e5e/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) your code in 2.796 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/atlp-rwanda/tech-strikers-bn-backend/837527c582d32c231319de91fab09efa44b1accc/atlp-rwanda/tech-strikers-bn-backend/7e55e2edc3dba77480ef22a969249d63f5609e5e/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=atlp-rwanda)

hervebutera commented:
> I thought changing profile, we had to add a picture and other information related to the user!

Now user information and a profile picture included can be updated.

hervebutera committed:
feat(user): add get and update user information by id routes

add function that gets user information by id from DB
add function that updates user information by id in DB
add multer middleware to handle image files for profile picture
add function that uploads picture to cloudinary

[delivers #175636847]
